### PR TITLE
feat: add CLI tests for angular-sitemap-generator functionality

### LINKED ISSUE #5: Insufficient Unit Test Coverage
**Description:**
The current unit tests provide very limited coverage of the library’s functionality. At the moment, they only test a single command and perform a minimal validation of its output. This approach was initially introduced to get the CI workflow up and running, but it does not adequately verify the correctness or reliability of the library.

**Expected Behavior:**
The test suite should be expanded to cover a broader range of features and use cases. Each major function and component of the library should have corresponding tests that validate both expected behavior and edge cases.

**Proposed Improvements:**
- Add unit tests for all public-facing commands and features
- Include edge case scenarios and error handling validation
- Improve assertions to ensure outputs are thoroughly validated

**Additional Context:**
Improving test coverage will increase confidence in the library, prevent regressions, and make future development safer and more maintainable.

### PR DESCRIPTION
**Pull request by KCTaka**
Fixes: #5 

This pull request adds a comprehensive CLI test suite for the project and updates the test scripts in `package.json` to support Node.js's native test runner. The main focus is on ensuring the CLI behaves correctly under various scenarios, including help/version output, argument validation, option handling, and error conditions.

**Test suite additions:**

* Added a new test file `test/index.cli.test.js` that uses Node.js's built-in `node:test` and `assert` modules to validate the CLI's behavior, including argument parsing, error handling, and generator invocation.
  * The test suite includes fixtures to create isolated environments, simulates the build process, and checks output files and exit codes for various CLI argument combinations.

**Build and test script updates:**

* Updated the `test` script in `package.json` to use `node --test`, and added a new `test:cli` script to specifically run the CLI test file.

### CODE REVIEW

File: test/index.cli.test.js
Comment (borisonekenobi) [inline test/index.cli.test.js:1]: Convert to TypeScript for type safety

File: test/index.cli.test.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+const os = require('node:os');
+const path = require('node:path');
+const {spawnSync} = require('node:child_process');
+const test = require('node:test');
+
+const repoRoot = path.resolve(__dirname, '..');
+const cliPath = path.join(repoRoot, 'index.js');
+const packageJson = require(path.join(repoRoot, 'package.json'));
+
+function createFixture(t, options = {}) {
```
Comment (borisonekenobi) [inline test/index.cli.test.js]: Please move this and all fixture related functions to their own file

File: test/index.cli.test.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  const {tscExitCode = 0} = options;
+
+  const cwd = fs.mkdtempSync(path.join(os.tmpdir(), 'asg-cli-test-'));
+  const binPath = path.join(cwd, 'bin');
+  fs.mkdirSync(binPath, {recursive: true});
+
+  // Provide a local "tsc" command so tests do not depend on the host toolchain.
```
Comment (borisonekenobi) [inline test/index.cli.test.js]: This is unneeded

**Review by borisonekenobi — CHANGES_REQUESTED** [review]

File: test/index.cli.test.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+  assert.equal(result.status, 1);
+  assert.match(result.stderr, /Build failed:/);
+  assert.equal(fs.existsSync(path.join(fixture.cwd, 'generator-ran.txt')), false);
+});
+
+test('fails when an option requiring a value is missing its argument', () => {
+  const result = runCli(['--path']);
+
+  assert.notEqual(result.status, 0);
```
[the remark this answers is not in the document]
Comment (KCTaka) [reply test/index.cli.test.js]: done

File: test/index.cli.test.js (hunk 1 above)
Comment (KCTaka) [reply test/index.cli.test.js]: done

File: test/index.cli.test.js (hunk 2 above)
Comment (KCTaka) [reply test/index.cli.test.js]: nah
